bibtex-completion.el

```diff
@@ -819,11 +825,15 @@ fields.  If FIELDS is empty, all fields are kept.  Also add a
 DO-NOT-FIND-PDF is non-nil, this function does not attempt to
 find a PDF file."
   (when entry ; entry may be nil, in which case just return nil
-    (let* ((fields (when fields (append fields (list "=type=" "=key=" "=has-pdf=" "=has-note="))))
+    (let* ((fields (when fields (append fields (list "=has-link" "=type=" "=key=" "=has-pdf=" "=has-note="))))
            ; Check for PDF:
            (entry (if (and (not do-not-find-pdf) (bibtex-completion-find-pdf entry))
                       (cons (cons "=has-pdf=" bibtex-completion-pdf-symbol) entry)
                     entry))
+           ; Check for link:
+           (entry (if (or (assoc "doi" entry) (assoc "url" entry))
```

Comment:
I think this line is wrong.

Reply:
Looks okay to me.  It should be true if either doi or url is defined.

Reply:
@bdarcus

It looks to me like you're kind of not doing anything there

Expanding the expression out gives:

```lisp
           (entry (if (or (assoc "doi" entry)
                          (assoc "url" entry)))
```

Which is an if condition that has no body... so it always evaluates to nil. 

I beleive the intention was to check whether the entry has a doi or url field and if so add it to the `entry` alist. In which case this should work:

```lisp
           (entry (if-let ((link (cdr (or (assoc "doi" entry)
                                          (assoc "url" entry)))))
                      (cons (cons "=link=" link) entry)
                    entry))

```

Reply:
Oh wait, I believe you were trying to go for something like `=has-pdf=` where the value is a symbol to render. In which case this is probably what you want:

```lisp
           (entry (if-let ((link (cdr (or (assoc "doi" entry)
                                          (assoc "url" entry)))))
                      (cons (cons "=has-link=" bibtex-completion-link-symbol) entry)
                    entry))
```

Reply:
So this is the current code:

``` elisp
           (entry (if (or (cdr (assoc "doi" entry)) (cdr (assoc "url" entry)))
                      (cons (cons "=has-link=" bibtex-completion-link-symbol) entry)
                    entry))
```

I intend that to be saying if the entry has a doi or url field, then add a "=has-link=" field (cons).

I was basing the code on the preceding one for pdfs, which obviously works, and was trying to be consistent with it.

Reply:
@bdarcus

Okay, looks good to me. What's the issue you wanted me to take a look at?

Reply:
I was just hoping fresh eyes would see something we have not so far!

I'll take another close look soon and see if I can figure this out. Maybe at that point I'll have a more specific question.

Thanks for checking!